ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -14101,6 +14107,110 @@ void SemanticErrorChecker::checkEmbedActivity(IHqlExpression * call)
     }
 }
 
+void SemanticErrorChecker::checkGrouping(IHqlExpression * expr)
+{
+    IHqlExpression * attrs = expr->queryChild(3);
+    if (attrs && queryAttributeInList(groupedAtom, attrs))
+        return;
+
+    IHqlExpression * groups = expr->queryChild(2);
```

Comment:
queryRealChild(expr, 2) avoids the extra tests on the next line  (You could also alternatively call queryDatasetGroupBy().)